### ISSUE TITLE
GOLD ecosystem in submissions vs in GOLD

### ISSUE THREAD

**Issue by mslarae13** [microbiomedata/issues#739]
Adina completed metadata via the submission portal for GLBRC see study here: https://data.microbiomedata.org/details/study/nmdc:sty-11-e4yb9z58

I am pretty sure this is this GOLD study : https://gold.jgi.doe.gov/study?id=Gs0128851
- https://genome.jgi.doe.gov/portal/Seadynanfunction/Seadynanfunction.info.html

Adina provided a DOE award, but on the data tab did not indicate that the metagenome data was generated at JGI in the submission.

In GOLD, the ecosystem path for biosamples is (spot checked, did not look at the ALL)
 
Ecosystem | Host-associated
-- | --
Ecosystem Category | Plants
Ecosystem Type | Phyllosphere
Ecosystem Subtype | Phylloplane/Leaf
Specific Ecosystem | Unclassified

In NMDC submission portal, the path provided is
<meta name="generator" content="Handsontable"/><style type="text/css">td{white-space:normal}br{mso-data-placement:same-cell}</style>
Environmental | Terrestrial | Plant-associated | Leaf | Phyllosphere
-- | -- | -- | -- | --

2 problems..
1. They don't match. What do we use.
2. Plant-associated | Leaf | Phyllosphere isn't valid. I'm not sure how it passed validation & got submitted!

@pkalita-lbl can you help figure out why this was valid and isn't now... I am guessing GOLD updated it's paths, and we pulled the updated paths? Cuz I'm looking at https://gold.jgi.doe.gov/ecosystemtree & these are not infact there... but IDK how Adina could've submitted if it wasn't valid before.

@aclum @sujaypatil96  can you check out GOLD and let me know if I am correct on the GOLD study
@aclum  can you chime in on what we should use? In the past, @emileyfadrosh has commented that we should use what the user says as it's their samples and their classification.. but, now it's not valid!

**Comment by pkalita-lbl** [comment]
My notes suggest that we brought that submission into Mongo in November of 2023. 

The [changes](https://github.com/microbiomedata/submission-schema/pull/173) to keep the submission schema in sync with GOLD ecosystem classification terms were done in January 2024. Those changes also updated the range for the 5 GOLD ecosystem classification slots. Prior to those changes, the range in anything _other than_ the soil template (and the submission in question uses the plant-associated template) was `string`.

So yeah it makes sense to me that the submission was valid in November 2023, but not now. And it's because of improvements to our schema, not because of changes at GOLD.

**Comment by mslarae13** [comment]
That makes sense! So in light of that
@aclum I suggest we submit a change-sheet to make the the NMDC GOLD paths match GOLD. Since that's actually valid.

**Comment by aclum** [comment]
Yes, this is the correct GOLD study ID. You can check this by getting the site award number from the osti page (https://www.osti.gov/award-doi-service/biblio/10.46936/10.25585/60000818) and using that ID to query the GOLD studies endpoint, you can do this interactively from the swagger UI if you are logged in with your ORCID. The json document that it returns has the GOLD study ID value in the field 'studyGoldId'. 

Do we have a mapping file between nmdc IDs and GOLD IDs? If so we can use the existing GOLD ingest code to pull update the the gold fields,  add the bioproject & sample information and create the omics records so we can pull the data in.

**Comment by mslarae13** [comment]
I am pretty sure the gold biosample IDs are stored. So yes?
@sujaypatil96 can you help with this?

**Comment by aclum** [comment]
GOLD biosample IDs are not stored for this project, `{'part_of':'nmdc:sty-11-e4yb9z58','gold_biosample_identifiers':{$exists:true}}` returns no records in mongo prod. Is this metadata in Adina's original submission @pkalita-lbl or do we need to go back to Adina for mapping information.

**Comment by pkalita-lbl** [comment]
There are no GOLD identifier slots in the submission schema. Unless the mapping between her sample names and GOLD identifiers is in an auxiliary file sitting in someone's inbox then we might need to go back to her for that information.

**Comment by mslarae13** [comment]
@aclum @sujaypatil96 anything left to do here? Your last comment seems mean this is resolved?

**Comment by aclum** [comment]
This is still outstanding. @sujaypatil96  please repull the GOLD ecosystem path slots and update the information if it doesn't match.

**Comment by sujaypatil96** [comment]
@aclum I wrote a script to repull the GOLD ecosystem path slot information onto the GLBRC biosamples.

I was doing some spot checking with examples and consider this biosample: `nmdc:bsm-11-0qmfcv11`

It currently has values like this:

```
      "ecosystem": "Environmental",
      "ecosystem_category": "Terrestrial",
      "ecosystem_subtype": "Leaf",
      "ecosystem_type": "Plant-associated",
      "specific_ecosystem": "Phyllosphere",
```

Turns out `nmdc:bsm-11-0qmfcv11` (like many other biosamples) has multiple GOLD biosample IDs asserted on it which are –
* `gold:Gb0157950`: https://gold.jgi.doe.gov/biosample?id=Gb0157950
* `gold:Gb0150590`: https://gold.jgi.doe.gov/biosample?id=Gb0150590

Now if you look at the ecosystem path values (in the API response) for both of these biosamples, you'll see a discrepancy like below:

["Field 'ecosystem_subtype': gold:Gb0157950='Unclassified' vs gold:Gb0150590='Phylloplane/Leaf'"]

So how is the script supposed to deal with this discrepancy? If it has to choose between 'Unclassified' and some value that's not 'Unclassified' then always pick the latter?

Note, the other ecosystem values have also changed:

* ecosystem –> 'Host-associated'
* ecosystem_category –> 'Plants'
* ecosystem_type –> 'Phyllosphere'
* 'specific_ecosystem' –> 'Unclassified'

**Comment by aclum** [comment]
I have created a JIRA ticket for Reddy asking for this to be cleaned up on the GOLD side. I'll assign this to myself until it is resolved.

**Comment by ssarrafan** [comment]
Blocked and @aclum is traveling for the next 2 week sprint. Removing from sprint board. Adding backlog label back.

**Comment by aclum** [comment]
This has still not been resolved and is unlikely to be. Closing as won't fix.
